src/Microsoft.AspNetCore.Mvc.Core/Internal/FilterActionInvoker.cs

```diff
@@ -179,7 +180,22 @@ public abstract class FilterActionInvoker : IActionInvoker
 
         private IFilterMetadata[] GetFilters()
         {
-            return _filterCache.GetFilters(Context);
+            if(_filters == null)
+            {
+                _filters =_controllerActionInvokerCache.GetFilters(Context);
+            }
+
+            return _filters;
+        }
```

Comment:
Don't do this. This should be called once and then accessed through the cursor later. This is only ever called once.